sigstore/sign.py

```diff
@@ -355,7 +409,10 @@ def _from_trust_config(cls, trust_config: ClientTrustConfig) -> SigningContext:
         signing_config = trust_config.signing_config
         return cls(
             fulcio=FulcioClient(signing_config.get_fulcio_url()),
-            rekor=RekorClient(signing_config.get_tlog_urls()[0]),
+            rekor=RekorClient(
+                signing_config.get_tlog_urls()[0],
+                signing_config._inner.rekor_tlog_urls[0].major_api_version,
+            ),
```

Comment:
my initial idea was to make get_tlog_urls() take a majorversion argument: so this code would do something like

```
logs=signing_config.get_tlogs(2)
if logs:
 rekor = Rekor2Client(v2_logs[0])
 else:
  logs = signing_config.get_tlogs(1)
  rekor = RekorClient(v2_logs[0])
 
```

obviously you can make it work with a single RekorClient as well, that's just to show the idea

Reply:
There is no `get_tlogs ()`, did you mean `get_tlog_urls()`?

That could work if we change `get_tlog_urls(self) -> list[str]` to also return a `v1.Service`.

`get_tlog_service(self, api_major_version: int) -> v1.Service`, and maybe also change the name to `get_tlog_service`, since we're currently only supporting `ServiceSelector.ANY`

Reply:
* Yes, I mean get_tlog_urls()
* returning a list of urls is enough in the proposal since this code knows which major version it asks for from get_log_urls()
* I think returning a list is correct -- SigningConfig does not support other serviceselectors **yet**, but we can fix that later and then the return value is a naturally a list

Reply:
Ok, and you prefer the caller of `get_tlog_urls(self, api_major_version: int) -> list[str]` prefer the known highest `api_major_version`?

Reply:
I see we're meant to use the highest that the client supports

https://github.com/sigstore/rekor-tiles/pull/255/files#diff-eb568acf84d583e4d3734b07773e96912277776bad39c560392aa33ea2cf2210R373-R375

Reply:
right: Once all public good instances have v2 running we can just drop support for v1 altogether IMO but for now I see it as the signing code responsibility to ask for the urls/services for a specific version in order of preference: this means that  all of the complexity of Services (like service selectors, operators etc) can remain an implementation detail of `trust.py` -- the signing code just knows which versions it supports and asks for urls.